src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -289,13 +289,17 @@ public void publish(Stream stream, Map<String, Object> payload, Date timestamp)
     }
 
     public void publish(Stream stream, Map<String, Object> payload, Date timestamp, String groupKeyHex) {
+        publish(stream, payload, timestamp, groupKeyHex, null);
+    }
+
+    public void publish(Stream stream, Map<String, Object> payload, Date timestamp, String groupKeyHex, String partitionKey) {
```

Comment:
Let's use the same order of arguments as in the js client?

```suggestion
    public void publish(Stream stream, Map<String, Object> payload, Date timestamp, String partitionKey, String groupKeyHex) {
```

For reference, JS client:

```
async publish(streamObjectOrId, data, timestamp = new Date(), partitionKey = null, groupKey) {
    ...
```